src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,114 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { getTodos } from '../../api/api';
+import { Todo } from '../../react-app-env';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+type Props = {
+  onHandlerUserId: (id: number) => void
+};
+
+export const TodoList: React.FC<Props> = ({ onHandlerUserId }) => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedOption, setSelectedOption] = useState('');
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const options = ['all', 'active', 'completed'];
+
+  useEffect(() => {
+    getTodos()
+      .then(todoItems => {
+        setAllTodos(todoItems);
+        setTodos(todoItems);
+      });
+  }, []);
+
+  const filteredByTitle = todos
+    .filter(todo => todo.title.toLowerCase().includes(title.toLowerCase()));
+
+  const visibleTodos = () => {
+    switch (selectedOption) {
+      case 'all':
+        setTodos(allTodos);
+        break;
+      case 'active':
+        setTodos([...allTodos].filter(todo => !todo.completed));
```

Comment:
you don't have to make an array copy, filter is immutable method 
Also, use functional setState here